provider/internal/builtin/service_client.go

```diff
@@ -132,52 +130,14 @@ func (p *builtintServiceClient) Evaluate(cap string, conditionInfo []byte) (prov
 			return response, fmt.Errorf("Could not parse provided xpath query '%s': %v", cond.XML.XPath, err)
 		}
 		//TODO(fabianvf): how should we scope the files searched here?
-		var xmlFiles []string
-		if len(cond.XML.Filepaths) == 0 {
```

Comment:
I think we want this branch to remain in the provider, since it's pretty specific to what the XML provider needs and isn't generally useful. The `else if` and `else` branches make sense to move out though